modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/RevokeOneTimeTokenFlowTestCase.java

```diff
@@ -0,0 +1,186 @@
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.xpath.XPathExpressionException;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * The test case for One Time Revocation flow. The test case will create an API and attach the revoke
+ * one time token policy. Then will test with JWT tokens that has the same scope claim as policy attribute scope
+ */
+public class RevokeOneTimeTokenFlowTestCase extends APIManagerLifecycleBaseTest {
+
+    private final String API_CONTEXT = "RevokeOneTimeTokenAPITest";
+    private String applicationId;
+    private String apiId;
+    private String accessToken;
+    private Map<String, String> policyMap;
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+
+        super.init();
+
+        //Create the API
+        HttpResponse applicationResponse = restAPIStore.createApplication(APPLICATION_NAME,
+                "Test Application RevokeOneTimeToken", APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED,
+                ApplicationDTO.TokenTypeEnum.JWT);
+        applicationId = applicationResponse.getData();
+        policyMap = restAPIPublisher.getAllCommonOperationPolicies();
+
+        String API_END_POINT_POSTFIX_URL = "xmlapi";
+        String apiEndPointUrl = getAPIInvocationURLHttp(API_END_POINT_POSTFIX_URL, API_VERSION_1_0_0);
+        String API_NAME = "RevokeOneTimeTokenAPITest";
+        APIRequest apiRequest = new APIRequest(API_NAME, API_CONTEXT, new URL(apiEndPointUrl));
+        apiRequest.setVersion(API_VERSION_1_0_0);
+        apiRequest.setTiersCollection(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setTier(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setTags(API_TAGS);
+        apiId = createPublishAndSubscribeToAPIUsingRest(apiRequest, restAPIPublisher, restAPIStore, applicationId,
+                APIMIntegrationConstants.API_TIER.UNLIMITED);
+
+        //Create the JWT access token with the scope OTT
+        List<String> grantTypes = new ArrayList<>();
+        grantTypes.add("client_credentials");
+        ArrayList<String> scopes = new ArrayList<>();
+        scopes.add("OTT");
+
+        ApplicationKeyDTO applicationKeyDTO = restAPIStore
+                .generateKeys(applicationId, "3600", null,
+                        ApplicationKeyGenerateRequestDTO.KeyTypeEnum.PRODUCTION, scopes, grantTypes);
+        assert applicationKeyDTO.getToken() != null;
+        accessToken = applicationKeyDTO.getToken().getAccessToken();
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Invoke the API before adding the revoke one time token policy")
+    public void testAPIInvocationBeforeAddingRevokeOneTimePolicy() throws Exception {
+
+        org.apache.http.HttpResponse invokeAPIResponse = invokeAPI();
+        assertEquals(invokeAPIResponse.getStatusLine().getStatusCode(), HTTP_RESPONSE_CODE_OK);
+    }
+
```

Comment:
You can add the new test for  Eg: testAPIInvocationOfAttachedRevokeOneTimePolicyUsingDefaultScopeToken

Reply:
You can use better name for the test cases :-)